src/main/java/com/codeup/adlister/controllers/EditPasswordServlet.java

```diff
@@ -0,0 +1,62 @@
+package com.codeup.adlister.controllers;
+
+import com.codeup.adlister.dao.DaoFactory;
+import com.codeup.adlister.models.User;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(name = "controllers.EditPasswordServlet", urlPatterns = "/editPassword")
+public class EditPasswordServlet extends HttpServlet {
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
+        if (request.getSession().getAttribute("user") == null) {
+            response.sendRedirect("/login");
+            return;
+        }
+
+        Long id = Long.parseLong(request.getParameter("id"));
+        User user = DaoFactory.getUsersDao().findById(id);
+        request.setAttribute("user", user);
+        request.getRequestDispatcher("/WEB-INF/editProfile.jsp").forward(request, response);
+
+    }
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        Long id = Long.parseLong(request.getParameter("id"));
+        String username = request.getParameter("username");
+        String email = request.getParameter("email");
+
+        // after a user submits the form..
+        // check if a username already exists in the database for the newly edited name.
+        // But also handle the issue of when a user doesn't change their name.
+
+        //get the existingUser in the database:
+        User existingUser = DaoFactory.getUsersDao().findById(id);
+        if (!existingUser.getUsername().equals(username)) {
```

Comment:
This is a recommendation for the future. Usually there's a page exclusively to change the password, and another to change anything else. This should be split 2 different pages in your capstone project. Another observation is that many applications out there won't let you change your username, which is another thing to consider :)